aten/src/ATen/TensorCompare.h

```diff
@@ -0,0 +1,100 @@
+#include "ATen/ATen.h"
+#include "ATen/Dispatch.h"
+#include "ATen/ExpandUtils.h"
+#include "ATen/NativeFunctions.h"
+
+
+template <template<typename> class, typename, typename, template<template<typename> class, typename, typename> class>
+struct CmpOpScalar {};
+
+// Comparators have special cases for integral tensors and floating scalars due to
+// the floating scalars being automatically cast to integral types during the comparison
+template<typename scalar_out, typename scalar, template<template<typename> class, typename, typename> class CmpOpImpl>
+struct CmpOpScalar<std::less, scalar_out, scalar, CmpOpImpl> {
+  static void apply(at::Tensor& ret, const at::Tensor& self, at::Scalar other) {
+    if (isIntegralType(self.type().scalarType()) && other.isFloatingPoint()) {
+      auto other_double = other.to<double>();
+      auto other_long = other.to<int64_t>();
+      if (other_double != other_long) {
+        other = at::Scalar(ceil(other_double));
```

Comment:
Don't we want to cast the `ceil(other_double)` to `int64_t` or sth? Won't this remain a floating point scalar if we leave it as is?

Reply:
It shouldn't matter because we're gonna cast the value to the correct type in the CmpOpImpl.